### PR TITLE
Add secret e2e tests for Windows

### PR DESCRIPTION
**Pull request by Kaderinho**
### What does this PR do?

Add `secret` e2e test for Windows

### Motivation

More coverage

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by clarkb7 — APPROVED** [review]
LGTM ty!

File: test/new-e2e/tests/agent-subcommands/secret/secret_nix_test.go [hunk 1]
```diff
@@ -3,42 +3,49 @@
...
-	e2e.Suite[e2e.AgentEnv]
+type linuxSecretSuite struct {
+	baseSecretSuite
 }
 
-func TestAgentSecretSuite(t *testing.T) {
-	e2e.Run(t, &agentSecretSuite{}, e2e.AgentStackDef())
+func TestLinuxSecretSuite(t *testing.T) {
+	t.Parallel()
+	e2e.Run(t, &linuxSecretSuite{}, e2e.AgentStackDef(e2e.WithVMParams(ec2params.WithOS(ec2os.UbuntuOS))), params.WithDevMode())
```
Comment (dustmop) [inline test/new-e2e/tests/agent-subcommands/secret/secret_nix_test.go:25]: Is `WithDevMode` supposed to be here? I thought it was only meant for developing tests, or is it used here to share state between the tests?

File: test/new-e2e/tests/agent-subcommands/secret/secret_win_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	assert.Contains(v.T(), output, "Executable path: C:\\Windows\\system32\\cmd.exe")
+	assert.Regexp(v.T(), "Executable permissions: error: invalid executable 'C:\\\\Windows\\\\system32\\\\cmd.exe': other users/groups than LOCAL_SYSTEM, .+ have rights on it", output)
+	assert.Regexp(v.T(), "Number of secrets .+: 0", output)
+}
+
+//go:embed fixtures/setup_secret.ps1
+var secretSetupScript []byte
+
+func (v *windowsSecretSuite) TestAgentSecretCorrectPermissions() {
+	config := `secret_backend_command: C:\secret.bat
```
Comment (dustmop) [inline test/new-e2e/tests/agent-subcommands/secret/secret_win_test.go:52]: These files are only created on the VM, not on the running user's local machine machine, right? If it's the latter, we should probably use `%TEMP%` or `%USERPROFILE%\AppData\Local\Temp` instead.

**Review by dustmop — APPROVED** [review]
Some small questions within, but LGTM aside from that.

File: test/new-e2e/tests/agent-subcommands/secret/secret_nix_test.go (hunk 1 above)
Comment (Kaderinho) [reply test/new-e2e/tests/agent-subcommands/secret/secret_nix_test.go:25]: You're right, I forgot to remove it mb 👍

File: test/new-e2e/tests/agent-subcommands/secret/secret_win_test.go (hunk 2 above)
Comment (Kaderinho) [reply test/new-e2e/tests/agent-subcommands/secret/secret_win_test.go:52]: > These files are only created on the VM, not on the running user's local machine machine, right?

Yes, these are created on the Windows VM where the Agent is installed (and which is destroyed at the end of the test suite)

**Pull request merged**